components/atom/slider/src/markerFactory.js

```diff
@@ -22,11 +22,30 @@ const linearMarksFactory = ({step, min, max, ticks}) => {
       }, {})
 }
 
-const markerFactory = ({step, min, max, marks}) => {
+const fullWidthMarksFactory = marks => {
+  const marksKeys = Object.keys(marks)
+  const firstMarkKey = marksKeys[0]
+  const lastMarkKey = marksKeys[marksKeys.length - 1]
+
+  const nextMarks = {
+    ...marks,
+    [firstMarkKey]: {label: marks[firstMarkKey], style: {transform: 'inherit'}},
+    [lastMarkKey]: {
+      label: marks[lastMarkKey],
+      style: {transform: 'inherit', left: 'auto', right: '0'}
+    }
```

Comment:
do you think we could achieve implement this in a pure CSS way? 

just as an idea, could we style `first-child` and `last-child` markers accordingly if the component has the `fullWidth` prop set to `true`?  

IMHO it'd be cleaner... what do you think?

Reply:
Yes! I'm with you with the approach, but here we're using `rc-slider` and is managing this styles with inline mode, so we only do that in CSS using `!important` declaration, instead them provide this `style` prop for each markers. For me, we could do an exception using `!important`, but It looks a bit tricky.

Reply:
oh... if the only alternative we have here is using `!important`, I think your original implementation is cleaner then